### PR TITLE
(BKR-1096) Block hosts flag with provision subcommand

### PR DESCRIPTION
**Pull request by rishijavia**
Removed hosts as a Thor class_option so it cannot be used by all subcommands. It is restricted to be used only with init command. Hosts is added as Thor method_option with init. Changed spec and acceptance tests accordingly.

### CODE REVIEW

File: acceptance/tests/subcommands/provision.rb [hunk 1]
```diff
@@ -10,8 +10,12 @@ def delete_root_folder_contents
     delete_root_folder_contents
     on(default, 'beaker init')
     result = on(default, 'beaker provision --hosts centos6-64')
+    assert_match(/ERROR/, result.raw_output)
+    on(default, 'beaker init --hosts centos6-64')
+    result = on(default, 'beaker provision')
     assert_match(/Using available host/, result.stdout)
 
+
```
Comment (kevpl) [inline acceptance/tests/subcommands/provision.rb]: @rishijavia is this just leftover?

Comment (rishijavia) [inline acceptance/tests/subcommands/provision.rb]: @kevpl yes, updating commit

**Comment by tvpartytonight** [comment]
@rishijavia this looks good; I think that if we are going to merge this work in, we should also make it so that `beaker init` requires a `--hosts` argument, otherwise users will be kind of stuck if the simply do `beaker init` without `--hosts`. I'll make a ticket for that work, it should be simply making `method_option` for hosts required for init.

**Comment by rishijavia** [comment]
test this please

**Comment by rishijavia** [comment]
test this please

File: acceptance/tests/subcommands/init.rb [hunk 2]
```diff
@@ -7,7 +7,8 @@ def delete_root_folder_contents
 
   step 'ensure beaker init writes YAML configuration files to disk' do
     delete_root_folder_contents
-    on(default, 'beaker init')
+    result = on(default, 'beaker init --hosts centos6-64')
+    assert_match(/Trying as beaker-hostgenerator input/, result.stdout)
```
Comment (tvpartytonight) [inline acceptance/tests/subcommands/init.rb]: I'm not clear why this assert is here; it seems to be testing that beaker is trying input as a hostgenerator string, but that is not what this step is trying to test; it's ensuring that YAML files are outputted to disk on the SUT. It seems like this assert is testing the implementation not pertinent to subcommand. 

A better test might be to read in the subcommand_options and ensure that the HOSTS object has a single element in the array.

File: acceptance/tests/subcommands/provision.rb [hunk 3]
```diff
@@ -10,8 +10,10 @@ def delete_root_folder_contents
     delete_root_folder_contents
     on(default, 'beaker init')
     result = on(default, 'beaker provision --hosts centos6-64')
+    assert_match(/ERROR/, result.raw_output)
```
Comment (tvpartytonight) [inline acceptance/tests/subcommands/provision.rb]: I'd like to ensure that the error message at least has some reference to the flag that is not allowed. Something like `/ERROR(.+)--hosts/` would work.

**Comment by rishijavia** [comment]
test this please

**Comment by tvpartytonight** [comment]
👍

**Pull request merged**